doc/vlan/SONiC_VLAN_trunk_all_except_none_HLD.md

```diff
@@ -0,0 +1,701 @@
+Q: A - Access (Untagged), T - Tagged
+NUM        Status      Q Ports
+20         Inactive    A  EthernetX
+
+```
+
+**Scenario 5:**
+<br> Vlan 20 does not exist.
+<br> Step 1: sonic(conf-if-EthernetX)#switchport trunk allowed vlan all
+<br> Step 2: sonic(conf-if-EthernetX)#switchport access vlan 20
+<br> Step 3: sonic(config)#interface vlan 20
+<br> Result: Port will be untagged in vlan 20
+<br> Verify: 'show run-configuration' output.  
+Sample show run-config output:
+```
+sonic# show running-configuration interface Ethernet X
+!
+interface EthernetX
+ mtu 9100
+ speed 40000
+ shutdown
+ switchport access Vlan 20
+ switchport trunk allowed Vlan 1-4094
+sonic#
+```
+
+**Scenario 6:**
+<br> Vlan 20 does not exist.
+<br> Step 1: sonic(conf-if-EthernetX)#switchport trunk allowed vlan except 20
+<br> Step 2: sonic(conf-if-EthernetX)#switchport access vlan 20
+<br> Step 3: sonic(config)#interface vlan 20
+<br> Result: Port will be untagged in vlan 20
+<br> Verify: 'show run-configuration' output.  
+Sample show run-config output:
+```
+sonic# show running-configuration interface Ethernet X
+!
+interface EthernetX
+ mtu 9100
+ speed 40000
+ shutdown
+ switchport access Vlan 20
+ switchport trunk allowed Vlan 1-19,21-4094
+sonic#
+```
+
+**Scenario 7:**
+<br> Step 1: sonic(conf-if-EthernetX)#switchport trunk allowed vlan except 5-7 
+<br> Step 2: sonic(conf-if-EthernetX)#switchport trunk allowed vlan add 5
+<br> Result: Port will be tagged in vlan 5
+<br> Verify: 'show run-configuration' output.  
+Sample show run-config output:
+```
+sonic# show running-configuration interface Ethernet X
+!
+interface EthernetX
+ mtu 9100
+ speed 40000
+ shutdown
+ switchport trunk allowed Vlan 1-5,8-4094
+sonic#
+```
+
+**Scenario 8:**
+<br> Vlan 20 does not exist.
+<br> Step 1: sonic(conf-if-EthernetX)#switchport trunk allowed vlan all
+<br> Step 3: sonic(config)#interface vlan 20
+<br> Result: Port tagged in vlan 1-4094
+<br> Step 3: sonic(conf-if-EthernetX)#switchport access vlan 20
+<br> Result: Port untagged in vlan 20
+<br> Verify: 'show run-configuration' output.  
+Sample show run-config output:
+```
+sonic# show running-configuration interface Ethernet X
+!
+interface EthernetX
+ mtu 9100
+ speed 40000
+ shutdown
+ switchport access Vlan 20
+ switchport trunk allowed Vlan 1-4094
+sonic#
+```
+
+**Scenario 9:**
+<br> Step 1: sonic(conf-if-EthernetX)#switchport trunk allowed vlan add 1-11 (no VLAN existing)
+<br> Step 2: sonic(conf-if-EthernetX)#switchport access vlan 11 (no VLAN existing)
+<br> Step 3: sonic(config)#interface range create vlan 1-11
+<br> Result: Port untagged in vlan 11
+<br> Step 4: sonic(conf-if-EthernetX)#no switchport access vlan
+<br> Result: Port will be tagged in vlan 1-11
+<br> Verify: 'show run-configuration' output.  
+Sample show run-config output:
+```
+sonic# show running-configuration interface Ethernet X
+!
+interface EthernetX
+ mtu 9100
+ speed 40000
+ shutdown
+ switchport trunk allowed Vlan 1-11
```

Comment:
should include the line "switchport access Vlan 11"

Reply:
No, since we gave "no switchport access vlan" at step 4